### PR TITLE
handle errors from health check watcher error channel

### PR DESCRIPTION
**Pull request by ahalimx86**
This fixes the coverity scan issue on error checking on error channel from health check watcher.

Signed-off-by: Abdul Halim <abdul.halim@intel.com>

### CODE REVIEW

File: pkg/utils/watcher.go [hunk 1]
```diff
@@ -62,6 +63,10 @@ func processEvents(w watcher) error {
 				return fmt.Errorf("error while waiting for the resource")
 			}
 			return nil
+		case err := <-errCh:
+			if err != nil {
+				return fmt.Errorf("watcher received error: %v", err)
```
Comment (dhruvpt) [inline pkg/utils/watcher.go]: Shouldn't we return explicit error code?

Comment (ahalimx86) [reply pkg/utils/watcher.go]: There is no error code. This error channel is used as signal for watcher threads to return in the event of any error encounter during grpc health check.

**Review by dhruvpt — APPROVED** [review]

**Review by nupurjai — CHANGES_REQUESTED** [review]
Test is failing with this. Check the runner errors.

**Comment by ahalimx86** [comment]
> Test is failing with this. Check the runner errors.

fixed unit tests

**Review by venkateshpnv — APPROVED** [review]

**Pull request merged**